File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/legacy/core/CameraState.kt
package dev.hebei.camerax_android.legacy.core

import dev.hebei.camerax_android.legacy.CameraState

val androidx.camera.core.CameraState.args
    get() = when (type) {
        androidx.camera.core.CameraState.Type.PENDING_OPEN -> CameraState.PENDING_OPEN
        androidx.camera.core.CameraState.Type.OPENING -> CameraState.OPENING
        androidx.camera.core.CameraState.Type.OPEN -> CameraState.OPEN
        androidx.camera.core.CameraState.Type.CLOSING -> CameraState.CLOSING
        androidx.camera.core.CameraState.Type.CLOSED -> CameraState.CLOSED
    }

File: camerax_android/lib/src/legacy/core/camera_state.dart
import 'package:camerax_android/src/legacy/camerax.g.dart' as $native;
import 'package:camerax_platform_interface/camerax_platform_interface.dart'
    as $interface;

extension NativeCameraState on $native.CameraState {
  $interface.CameraState get args {
    switch (this) {
      case $native.CameraState.pendingOpen:
        return $interface.CameraState.pendingOpen;
      case $native.CameraState.opening:
        return $interface.CameraState.opening;
      case $native.CameraState.open:
        return $interface.CameraState.open;
      case $native.CameraState.closing:
        return $interface.CameraState.closing;
      case $native.CameraState.closed:
        return $interface.CameraState.closed;
    }
  }
}
